cmd/stencil/lockfile_prune.go

```diff
@@ -0,0 +1,53 @@
+// Copyright (C) 2024 stencil contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"github.com/pkg/errors"
+	"github.com/urfave/cli/v2"
+	"go.rgst.io/stencil/pkg/slogext"
+	"go.rgst.io/stencil/pkg/stencil"
+)
+
+// NewLockfilePruneCommand returns a new urfave/cli.Command for the
+// lockfile prune command.
+func NewLockfilePruneCommand(log slogext.Logger) *cli.Command {
+	return &cli.Command{
+		Name:        "prune",
+		Description: "Prunes any non-existent files from the lockfile (will recreate any file.Once files on next run)",
+		Args:        true,
+		ArgsUsage: "[filename] [filename 2] [...] - if any names are passed, prune only checks passed filenames for " +
+			"pruning instead of all",
+		Action: func(c *cli.Context) error {
+			l, err := stencil.LoadLockfile("")
+			if err != nil {
+				return errors.Wrap(err, "failed to load lockfile")
+			}
+
+			prunedList := l.Prune(c.Args().Slice())
```

Comment:
It'd probably be better to use a varidiac (I think they're called Slice args in urfave/cli - basically you can add more than one of the same flag) flag argument instead of using positional arguments. The only reason I say this is because if you commit it the way it is now you can't change it later (what if you end up wanting subcommands of prune - now you have to breaking change release).

Reply:
fair nuff